centrosome.filter ZeroDivisionError

```
__________________________________________________________________________________________________________________ TestLineIntegration.test_01_02_two_lines __________________________________________________________________________________________________________________

self = <test_filter.TestLineIntegration testMethod=test_01_02_two_lines>

    def test_01_02_two_lines(self):
        img = np.ones((20,30)) * .5
        img[8,10:20] = 1
        img[12,10:20] = 0
>       result = F.line_integration(img, 0, 1, 0)

tests/test_filter.py:1770:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
centrosome/filter.py:1189: in line_integration
    smoothed = scind.gaussian_filter1d(rotated, sigma)
/usr/local/lib/python2.7/site-packages/scipy/ndimage/filters.py:271: in gaussian_filter1d
    weights = _gaussian_kernel1d(sigma, order, lw)[::-1]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

sigma = 0, order = 0, radius = 0

    def _gaussian_kernel1d(sigma, order, radius):
        """
        Computes a 1D Gaussian convolution kernel.
        """
        if order < 0:
            raise ValueError('order must be non-negative')
>       p = numpy.polynomial.Polynomial([0, 0, -0.5 / (sigma * sigma)])
E       ZeroDivisionError: float division by zero

/usr/local/lib/python2.7/site-packages/scipy/ndimage/filters.py:207: ZeroDivisionError
_______________________________________________________________________________________________________________ TestLineIntegration.test_01_03_diagonal_lines ________________________________________________________________________________________________________________

self = <test_filter.TestLineIntegration testMethod=test_01_03_diagonal_lines>

    def test_01_03_diagonal_lines(self):
        img = np.ones((20,30)) * .5
        i,j = np.mgrid[0:20,0:30]
        img[(i == j-3) & (i <= 15)] = 1
        img[(i == j + 3)] = 0
        expected = np.zeros((20,30), bool)
        expected[(i >= j-3) & (i <= j+3)] = True
>       result = F.line_integration(img, -45, 1, 0)

tests/test_filter.py:1784:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
centrosome/filter.py:1189: in line_integration
    smoothed = scind.gaussian_filter1d(rotated, sigma)
/usr/local/lib/python2.7/site-packages/scipy/ndimage/filters.py:271: in gaussian_filter1d
    weights = _gaussian_kernel1d(sigma, order, lw)[::-1]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

sigma = 0, order = 0, radius = 0

    def _gaussian_kernel1d(sigma, order, radius):
        """
        Computes a 1D Gaussian convolution kernel.
        """
        if order < 0:
            raise ValueError('order must be non-negative')
>       p = numpy.polynomial.Polynomial([0, 0, -0.5 / (sigma * sigma)])
E       ZeroDivisionError: float division by zero

/usr/local/lib/python2.7/site-packages/scipy/ndimage/filters.py:207: ZeroDivisionError
____________________________________________________________________________________________________________________ TestLineIntegration.test_01_04_decay ____________________________________________________________________________________________________________________

self = <test_filter.TestLineIntegration testMethod=test_01_04_decay>

    def test_01_04_decay(self):
        img = np.ones((25,23)) * .5
        img[10,10] = 1
        img[20,10] = 0
>       result = F.line_integration(img, 0, .9, 0)

tests/test_filter.py:1792:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
centrosome/filter.py:1189: in line_integration
    smoothed = scind.gaussian_filter1d(rotated, sigma)
/usr/local/lib/python2.7/site-packages/scipy/ndimage/filters.py:271: in gaussian_filter1d
    weights = _gaussian_kernel1d(sigma, order, lw)[::-1]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

sigma = 0, order = 0, radius = 0

    def _gaussian_kernel1d(sigma, order, radius):
        """
        Computes a 1D Gaussian convolution kernel.
        """
        if order < 0:
            raise ValueError('order must be non-negative')
>       p = numpy.polynomial.Polynomial([0, 0, -0.5 / (sigma * sigma)])
E       ZeroDivisionError: float division by zero

/usr/local/lib/python2.7/site-packages/scipy/ndimage/filters.py:207: ZeroDivisionError
```

**SciPy version:**
```
$ pip freeze | grep scipy
scipy==1.0.0
```

--- a/centrosome/filter.py
+++ b/centrosome/filter.py
@@ -690,11 +690,11 @@ def enhance_dark_holes(image, min_radius, max_radius, mask=None):
             smoothed_image = np.maximum(smoothed_image,output_image)
         previous_reconstructed_image = reconstructed_image
     return smoothed_image
-    
-    
+
+
 def granulometry_filter(image, min_radius, max_radius, mask=None):
     '''Enhances bright structures within a min and max radius using a rolling ball filter
-    
+
     image - grayscale 2-d image
     radii - a vector of radii: we enhance holes at each given radius
     '''
@@ -712,15 +712,15 @@ def granulometry_filter(image, min_radius, max_radius, mask=None):
     selected_granules_image = np.zeros(image.shape)
     #
     # Select granules by successive morphological openings
-    #    
+    #
     for i in range(max_radius+1):
         eroded_image = grey_erosion(eroded_image, mask=mask, footprint = se)
         opened_image = grey_dilation(eroded_image, inverted_image, footprint = se)
-        output_image = previous_opened_image - opened_image 
+        output_image = previous_opened_image - opened_image
         if i >= min_radius:
             selected_granules_image = np.maximum(selected_granules_image, output_image)
-        previous_opened_image = opened_image 
-    return selected_granules_image 
+        previous_opened_image = opened_image
+    return selected_granules_image
 
 def circular_average_filter(image, radius, mask=None):
     '''Blur an image using a circular averaging filter (pillbox)
@@ -1186,7 +1186,7 @@ def line_integration(image, angle, decay, sigma):
     #
     # Smooth in only the i direction
     #
-    smoothed = scind.gaussian_filter1d(rotated, sigma)
+    smoothed = scind.gaussian_filter1d(rotated, sigma) if sigma > 0 else rotated
     #
     # We want img_out[:,j+1] to be img_out[:,j] * decay + img[j+1]
     # Could be done by convolution with a ramp, maybe in FFT domain,

